applications/dashboard/controllers/api/AddonsApiController.php

```diff
@@ -0,0 +1,210 @@
+
+        return $r;
+    }
+
+    /**
+     * Query the available addons.
+     *
+     * @param array $query The query string.
+     * @return array Returns an array of addon information.
+     */
+    public function index(array $query) {
+        $this->permission('Garden.Settings.Manage');
+
+        $in = $this->schema([
+            'type:s?' => [
+                'description' => 'The type of addon.',
+                'enum' => ['addon', 'theme', 'locale']
+            ],
+            'enabled:b?' => 'Filter enabled or disabled addons.',
+            'themeType:s?' => [
+                'description' => 'Which theme to show the enabled status for.',
+                'enum' => ['desktop', 'mobile'],
+                'default' => 'desktop'
+            ]
+        ], 'in')->setDescription('List addons.');
+        $out = $this->schema([':a' => $this->fullSchema()], 'out');
+
+        $query = $in->validate($query);
+
+        $addons = $this->addonModel->getWhere($query);
+
+        $themeType = empty($query['themeType']) ? 'desktop' : $query['themeType'];
+        $addons = array_map(function (&$row) use ($themeType) {
+            return $this->filterOutput($row, $themeType);
+        }, $addons);
+        usort($addons, function ($a, $b) {
+            return strcasecmp($a['name'], $b['name']);
+        });
+        $addons = $out->validate($addons);
+
+        return $addons;
+    }
+
+    /**
+     * Get a single addon.
+     *
+     * @param string $addonID The addon ID to get.
+     * @param array $query The query string.
+     * @return array Returns an addon info row.
+     * @throws \Garden\Web\Exception\NotFoundException Throws an exception when the addon is not found.
+     */
+    public function get($addonID, array $query) {
+        $this->permission('Garden.Settings.Manage');
+
+        $id = $this->schema([
+            'addonID:s' => 'The ID of the addon.'
+        ], 'in')->setDescription('Get an addon.');
+
+        $in = $this->schema([
+            'themeType:s?' => [
+                'description' => 'Which theme to show the enabled status for.',
+                'enum' => ['desktop', 'mobile'],
+                'default' => 'desktop'
+            ]
+        ], 'in');
+        $out = $this->schema($this->fullSchema(), 'out');
+        $query = $in->validate($query);
+
+        $addons = $this->addonModel->getWhere(['addonID' => $addonID]);
+        if (empty($addons)) {
+            throw new \Garden\Web\Exception\NotFoundException("Addon");
+        }
+
+        $row = $out->validate($this->filterOutput(reset($addons)));
+        return $row;
+    }
+
+    /**
+     * Enable or disable an addon.
+     *
+     * @param string $addonID The addon ID.
+     * @param array $body The request body.
+     * @return array Returns an array of affected addons.
+     * @throws \Garden\Web\Exception\NotFoundException Throws an exception when the addon is not found.
+     */
+    public function patch($addonID, array $body) {
```

Comment:
If this is the extent of the planned functionality, I think it would make more sense as a subresource action. No fields beyond the "enabled" attribute can be modified. This is essentially a toggle. We can treat the "enabled" state as a subresource (e.g. PATCH /api/v2/addons/:id/enabled or PUT /api/v2/addons/:id/enabled if you aren't a stickler for idempotency).

Reply:
I see your point, but this is symmetric with the get endpoints.